vassal-app/src/main/java/VASSAL/build/module/map/boardPicker/board/mapgrid/PolygonEditor.java

```diff
@@ -434,6 +539,18 @@ public void mouseDragged(MouseEvent e) {
         polygon.xpoints[1] = polygon.xpoints[2] = e.getX();
         polygon.ypoints[2] = polygon.ypoints[3] = e.getY();
         repaint();
+        if (myConfigurer != null) {
+          myConfigurer.updateCoords(polygon);
+
+          final String s = polygon.xpoints[0] +
+            "," +
+            polygon.ypoints[0] +
+            " => " +
+            polygon.xpoints[2] +
+            "," +
+            polygon.ypoints[2];
+          myConfigurer.updateCoord(s);
+        }
```

Comment:
Two suggestions:

* Try making a polygon with a _lot_ of vertices. It doesn't take long before they run off the edge of the window. It would be better if they wrapped. There's a FlowSomethingOrOther component that we use in a few places to get a label that word-wraps.
* A big list of coordinates might be easier to read if each pair has parentheses around it.

Reply:
I've fixed both of these, it just happens at a different place in the code. This part here is the one that deals with the drawing-the-initial-square part.